server/filestore.go

```diff
@@ -3509,7 +3509,7 @@ func (fs *fileStore) removeMsg(seq uint64, secure, viaLimits, needFSLock bool) (
 	}
 
 	cb := fs.scb
-	fs.mu.Unlock()
+	fsUnlock()
```

Comment:
It's worth calling fsLock() on `3524` for consistency and to avoid introducing future bugs should the `fsLock` and `fsUnlock` pair get new behaviors, one would easily assume updating those functions would update the entire func behavior.

Reply:
Oh, I see the logic there is actually different. So ignore this comment sorry.